common/src/main/java/com/wynntils/models/skillpoint/SavableSkillPointSet.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.skillpoint;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class SavableSkillPointSet {
```

Comment:
This class really wants to be a record, doesn't it? You can create factory methods instead of the constructors.